trieste/acquisition/function/active_learning.py

```diff
@@ -245,3 +246,78 @@ def acquisition(x: TensorType) -> TensorType:
         return criterion
 
     return acquisition
+
+class BayesianActiveLearningByDisagreement(SingleModelAcquisitionBuilder):
+    """
+    Builder for the *Bayesian Active Learning By Disagreement* acquisition function defined in
+    :cite:`houlsby2011bayesian`. The acquisition function computes the information gain
+    of the predictive entropy.  
+
+    This acquisition function is intended to use for Binary Gaussian Process Classification
+    model with Bernoulli likelihood. integrating over nuisance parameters is currently not 
+    supported (see equation 6 of the paper)
+    """
+
+    def __init__(self, jitter: float = DEFAULTS.JITTER) -> None:
+        """
+        :param jitter: The size of the jitter to avoid numerical problem caused by the
+                log operation if variance is close to zero.
+        """
+        self._jitter = jitter
+
+    def __repr__(self) -> str:
+        """"""
+        return f"BayesianActiveLearningByDisagreement(jitter={self._jitter!r})"
+
+    def prepare_acquisition_function(
+        self,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param model: The model.
+        :param dataset: Unused.
+
+        :return: The determinant of the predictive function.
+        """
+
+        return bayesian_active_learning_by_disagreement(model, self._jitter)
+
+    def update_acquisition_function(
+        self,
+        function: AcquisitionFunction,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param function: The acquisition function to update.
+        :param model: The model.
+        :param dataset: Unused.
+        """
+        return function  # no need to update anything
+
+
+class bayesian_active_learning_by_disagreement(AcquisitionFunctionClass):
+    def __init__(self, model: ProbabilisticModel, jitter: float):
+        """
+
+        :param model: The model of the objective function.
+        :param jitter: The size of the jitter to avoid numerical problem caused by the
+                log operation if variance is close to zero.
+        :return: The Bayesian Active Learning By Disagreement acquisition function.
+        """
+        self._model = model
+        self._jitter = jitter
+
+    # @tf.function
+    def __call__(self, x: TensorType) -> TensorType:
```

Comment:
this is not supporting batches, right? lets assert here that x doesn't contain batches (there are other acq fncs that make a similar check) - we'll need a test that raises this exception as well

Reply:
there is also BatchBALD in https://proceedings.neurips.cc/paper/2019/file/95323660ed2124450caaac2c46b5ed90-Paper.pdf, but not sure if the code can be easily extended here...

Reply:
no it doesnt. will check it